app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -34,13 +62,108 @@ class MainActivity : BaseTeaserActivity() {
 
         VisionSafetyManager.create(VisionManager)
         VisionSafetyManager.visionSafetyListener = visionSafetyListener
+        return true
+    }
+
+    private fun initVisionManagerReplay(visionView: VisionView, sessionPath: String): Boolean {
+        if (sessionPath.isEmpty()) {
+            return false
+        }
+
+        this.sessionPath = sessionPath
+        VisionReplayManager.create(sessionPath)
+        VisionReplayManager.visionEventsListener = visionEventsListener
+        VisionReplayManager.start()
+        VisionReplayManager.setModelPerformance(modelPerformance)
+        visionView.setVisionManager(VisionReplayManager)
 
+        VisionSafetyManager.create(VisionReplayManager)
+        VisionSafetyManager.visionSafetyListener = visionSafetyListener
         return true
     }
 
+
+    private fun showReplayModeFragment() {
+        supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.fragment_container, ReplayModeFragment.newInstance(BASE_SESSION_PATH), ReplayModeFragment.TAG)
+                .addToBackStack(ReplayModeFragment.TAG)
+                .commit()
+        hideDashboardView()
+
+    }
+
+    private fun hideDashboardView() {
+        dashboard_container.hide()
+        title_teaser.hide()
+    }
+
+    private fun showDashboardView() {
+        dashboard_container.show()
+        title_teaser.show()
+    }
+
     override fun destroyVisionManager() {
+        return when (mode) {
```

Comment:
nit : ` = when` is a bit shorter, here and rest of funs

Reply:
code changed in next branches